src/extension/features/general/pos-style-currency-entry-mode/index.js

```diff
@@ -0,0 +1,124 @@
+import { create, all } from 'mathjs';
+import { Feature } from 'toolkit/extension/features/feature';
+import { componentLookup } from 'toolkit/extension/utils/ember';
+
+export class POSStyleCurrencyEntryMode extends Feature {
+  constructor() {
+    super();
+
+    this.mathEvaluatorInstance = null;
+    this.currencyFormatter = null;
+    this.accountCurrency = null;
+    this.decimalDigits = null;
+  }
+
+  shouldInvoke() {
+    const {
+      currencyFormatter,
+    } = ynab.YNABSharedLibWebInstance.firstInstanceCreated.formattingManager;
+
+    this.currencyFormatter = currencyFormatter;
+    this.accountCurrency = currencyFormatter.getCurrency();
+    this.decimalDigits = this.accountCurrency.decimal_digits;
+
+    // When 0, there are no currency sub-units => POS style entry is unneeded
+    return this.decimalDigits > 0;
+  }
+
+  invoke() {
+    this.wrapCurrencyInput('ynab-new-currency-input');
```

Comment:
The old `currency-input` is not wrapped as it seems to have a dead-code `commitValue`, but itself is only(?) used in reconciliation pop-over, thus doesn't seem to much of an impact for consistent UX.